network/hostnetwork/dht.go

```diff
@@ -1365,3 +1362,70 @@ func (dht *DHT) AddPossibleRelayID(id string) {
 func (dht *DHT) GetOriginHost() *host.Origin {
 	return dht.origin
 }
+
+func (dht *DHT) AddUncheckedNode(hostID id.ID, nonce []byte, ref core.RecordRef) {
+	unchecked := UncheckedNode{Ref: ref, Nonce: nonce}
+	dht.uncheckedNodes[hostID.String()] = unchecked
```

Comment:
any time limit to store unchecked nodes? we should clean this map from time to time